### PR TITLE
[Nokia ixs7215] Platform API 2.0 improvements

### PR DESCRIPTION
**Pull request by dflynn-Nokia**
This PR provides the following platform API 2.0 improvements for the Nokia
ixs7215 platform.

- add support for Fan and PSU eeproms
- bug fixes in platform firmware status and update logic
- unit test enhancements and bug fixes
- add platform specific reboot script
- relocate pcie.yaml file

#### Why I did it
This PR has been created for the following reasons
- Improve sonic-mgmt platform test suite pass rate
- Improve coverage of platform unit tests
- Provide platform specific reboot logic as per platform porting guide
- Fix bug due to pcie.yaml file being located in the wrong directory
 
#### How I did it
Various code changes to platform API 2.0 logic

#### How to verify it
Verify all 'show platform' commands work as expected
Verify pass rate of sonic-mgmt platform test suite

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [x] 202012

### CODE REVIEW

File: platform/marvell-armhf/sonic-platform-nokia/7215/sonic_platform/component.py [hunk 1]
```diff
@@ -34,6 +35,7 @@ class Component(ComponentBase):
         ["System-CPLD", "Used for managing SFPs, LEDs, PSUs and FANs "],
         ["U-Boot", "Performs initialization during booting"],
     ]
+    CPLD_UPDATE_COMMAND = 'cp /usr/sbin/vme /tmp; cp {} /tmp; cd /tmp; ./vme {};  '
```
Comment (jleveque) [inline platform/marvell-armhf/sonic-platform-nokia/7215/sonic_platform/component.py:38]: Delete trialing spaces in string

**Review by jleveque — CHANGES_REQUESTED** [review]

File: platform/marvell-armhf/sonic-platform-nokia/7215/sonic_platform/component.py (hunk 1 above)
Comment (dflynn-Nokia) [reply platform/marvell-armhf/sonic-platform-nokia/7215/sonic_platform/component.py]: Done

**Review by jleveque — CHANGES_REQUESTED** [review]
Please fix LGTM alerts. Note that the "Wrong name for an argument in a class instantiation" may be a false alarm if you are overriding inherited methods. Please let me know if you have any questions.

**Comment by dflynn-Nokia** [comment]
Thanks Joe. The multiply defined variable is an error on my part which I will fix in a new commit.
The 2 import errors are due to the __init__.py changes I made to model our code after that provided by Nvidia.
Our original __init__.py was empty. I could use your guidance on this one.
Lastly the "wrong name" issue is my attempt to pass values to our Eeprom class initializer to indicate whether the eeprom device is associated with a fan or psu. Other vendors seem to also use this approach.

**Comment by jleveque** [comment]
> The 2 import errors are due to the **init**.py changes I made to model our code after that provided by Nvidia.
> Our original **init**.py was empty. I could use your guidance on this one.

Since other vendors have already merged code with these approaches, I an OK with allowing these alerts for the time being. However, we eventually need to clean this up. I have not investigated what can be done to eliminate these, but different vendors have taken different approaches to creating their sonic-platform package. I would suggest looking at the different implementations and see if you can find an approach which eliminates these alerts.

> Lastly the "wrong name" issue is my attempt to pass values to our Eeprom class initializer to indicate whether the eeprom device is associated with a fan or psu. Other vendors seem to also use this approach.

Again, I am OK with allowing this alert for now. This is more of a false alarm because there are multiple sonic-platform packages, some which override and some which don't, and LGTM compares to the wrong definition. In the future, we may want to look into adding these parameters with default values in the base class or finding a different approach.

**Review by jleveque — APPROVED** [review]

**Comment by dflynn-Nokia** [comment]
Thanks Joe.
Any plan to allow PR authors to do this?

**Comment by jleveque** [comment]
> Thanks Joe.
> Any plan to allow PR authors to do this?

I'd love to be able to if possible. We're looking into it.

**Comment by dflynn-Nokia** [comment]
Can we get this one merged and backported to 202012?

**Pull request merged**